public/tier1/convar.h

```diff
@@ -464,251 +337,244 @@ friend class ConCommandHandle;
 	
-	// Call this function when ConVar changes
-	CUtlVector< FnChangeCallback_t > m_fnChangeCallbacks;
-#endif
-#endif // CONVAR_WORK_FINISHED
-	const char* m_pszName;
-	CVValue_t* m_cvvDefaultValue;
-	CVValue_t* m_cvvMinValue;
-	CVValue_t* m_cvvMaxValue;
-	const char* m_pszHelpString;
-	EConVarType m_eVarType;
-
-	// This gets copied from the ConVarDesc_t on creation
-	short unk1;
-
-	unsigned int timesChanged;
-	int64 flags;
-	unsigned int callback_index;
-
-	// Used when setting default, max, min values from the ConVarDesc_t
-	// although that's not the only place of usage
-	// flags seems to be:
-	// (1 << 0) Skip setting value to split screen slots and also something keyvalues related
-	// (1 << 1) Skip setting default value
-	// (1 << 2) Skip setting min/max values
-	int allocation_flag_of_some_sort;
-
-	CVValue_t** values;
-};
+	inline const T&	GetValue( int index = 0 ) const			{ return m_ConVar->GetValue( index ); }
+	inline void		SetValue(const T& value, int index = 0)	{ m_ConVar->SetValue( value, index ); }
 
+	inline bool		IsFlagSet( int64_t flag ) const		{ return m_ConVar->IsFlagSet( flag ); }
+	inline void		AddFlags( int64_t flags )			{ m_ConVar->AddFlags( flags ); }
+	inline void		RemoveFlags( int64_t flags )		{ return m_ConVar->RemoveFlags( flags ); }
+	inline int64_t	GetFlags( void ) const				{ return m_ConVar->GetFlags( ); }
 
-#ifdef CONVAR_WORK_FINISHED
-//-----------------------------------------------------------------------------
-// Purpose: Return ConVar value as a float
-// Output : float
-//-----------------------------------------------------------------------------
-FORCEINLINE_CVAR float ConVar::GetFloat( void ) const
-{
-	return m_pParent->m_Value.m_fValue;
-}
+private:
+	// sub_10B7BC0
+	void Init(ConVarHandle defaultHandle, EConVarType type)
+	{
+		this->m_Handle.Invalidate();
+		this->m_ConVar = nullptr;
 
-//-----------------------------------------------------------------------------
-// Purpose: Return ConVar value as an int
-// Output : int
-//-----------------------------------------------------------------------------
-FORCEINLINE_CVAR int ConVar::GetInt( void ) const 
-{
-	return m_pParent->m_Value.m_nValue;
-}
+		// qword_191A3D8
+		if (g_pCVar && (this->m_ConVar = g_pCVar->GetConVar<T>(defaultHandle)) == nullptr)
+		{
+			this->m_ConVar = ConVar_Invalid<T>();
+			// technically this
+			//result = *(char ***)(sub_10B7760((unsigned int)a3) + 80);
+		}
+		this->m_Handle = defaultHandle;
+	}
 
-//-----------------------------------------------------------------------------
-// Purpose: Return ConVar value as a color
-// Output : Color
-//-----------------------------------------------------------------------------
-FORCEINLINE_CVAR Color ConVar::GetColor( void ) const 
-{
-	unsigned char *pColorElement = ((unsigned char *)&m_pParent->m_Value.m_nValue);
-	return Color( pColorElement[0], pColorElement[1], pColorElement[2], pColorElement[3] );
-}
+	// sub_10B7C70

```

Comment:
These comments pointing to some ida funcs and refs aren't really much of a use since they'd get obsoleted quite quickly with newer binaries.